xbmc/settings/dialogs/GUIDialogContentSettings.cpp

```diff
@@ -89,46 +93,85 @@ bool CGUIDialogContentSettings::OnMessage(CGUIMessage &message)
-          // For now, we just close the dialog + send the GetPath() to open the addons window
-          std::string content = m_vecItems->Get(iSelected)->GetPath().substr(14);
-          OnCancel();
-          Close();
-          CBuiltins::GetInstance().Execute("ActivateWindow(AddonBrowser,addons://all/xbmc.metadata.scraper." + content + ",return)");
-          return true;
+        std::vector<std::pair<std::string, int>> labels;
+        if (m_content == CONTENT_ALBUMS || m_content == CONTENT_ARTISTS)
+        {
+          labels.push_back(std::make_pair(ADDON::TranslateContent(m_content, true), m_content));
         }
+        else
+        {
+          labels.push_back(std::make_pair(ADDON::TranslateContent(CONTENT_NONE, true), CONTENT_NONE));
+          labels.push_back(std::make_pair(ADDON::TranslateContent(CONTENT_MOVIES, true), CONTENT_MOVIES));
+          labels.push_back(std::make_pair(ADDON::TranslateContent(CONTENT_TVSHOWS, true), CONTENT_TVSHOWS));
+          labels.push_back(std::make_pair(ADDON::TranslateContent(CONTENT_MUSICVIDEOS, true), CONTENT_MUSICVIDEOS));
+        }
+        std::sort(labels.begin(), labels.end());
+
+        CGUIDialogSelect *dialog = (CGUIDialogSelect *)g_windowManager.GetWindow(WINDOW_DIALOG_SELECT);
+        if (dialog)
+        {
+          dialog->SetHeading(CVariant{ 20344 }); //Label "This directory contains"
+
+          int iIndex = 0;
+          int iSelected = 0;
+          for (const auto &label : labels)
+          {
+            dialog->Add(label.first);
+
+            if (m_content == label.second)
+              iSelected = iIndex;
+            iIndex++;
+          }
 
-        AddonPtr last = m_scraper;
-        m_scraper = std::dynamic_pointer_cast<CScraper>(m_scrapers[m_content][iSelected]);
-        m_lastSelected[m_content] = m_scraper;
+          dialog->SetSelected(iSelected);
+
+          dialog->Open();
+          // Selected item has not changes - in case of cancel or the user selecting the same item
+          if (dialog->GetSelectedLabel() == iSelected)
+            return true;
+          for (const auto &label : labels)
+          {
+            if (dialog->GetSelectedLabelText() == label.first)
+            {
+              m_content = static_cast<CONTENT_TYPE>(label.second);
+              break;
+            }
+          }
+
+          AddonPtr scraperAddon;
+          CAddonMgr::GetInstance().GetDefault(ADDON::ScraperTypeFromContent(m_content), scraperAddon);
+          m_scraper = std::dynamic_pointer_cast<CScraper>(scraperAddon);
 
-        if (m_scraper != last)
           SetupView();
+          SET_CONTROL_LABEL2(CONTROL_CONTENT_TYPE_BUTTON, dialog->GetSelectedLabelText());
+          SET_CONTROL_FOCUS(CONTROL_CONTENT_TYPE_BUTTON, 0);
+        }
+      }
+      else if (iControl == CONTROL_SCRAPER_LIST_BUTTON)
+      {
+        ADDON::TYPE type = ADDON::ScraperTypeFromContent(m_content);
+        std::string selectedAddonId;
```

Comment:
Set it to the "old" id. the second parameter is in/out. it will highlight the ones you pass in.